tools/provision/formula/cmake.rb

```diff
@@ -35,36 +32,20 @@ def install
       --mandir=/share/man
       --system-zlib
       --system-bzip2
+      --system-liblzma
     ]
 
-    # https://github.com/Homebrew/legacy-homebrew/issues/45989
-    if MacOS.version <= :lion
-      args << "--no-system-curl"
-    else
-      args << "--system-curl"
-    end
-
-    if build.with? "docs"
-      # There is an existing issue around OS X & Python locale setting
-      # See https://bugs.python.org/issue18378#msg215215 for explanation
-      ENV["LC_ALL"] = "en_US.UTF-8"
-      args << "--sphinx-man" << "--sphinx-build=#{Formula["sphinx-doc"].opt_bin}/sphinx-build"
-    end
-
+    # Keep these for initial compiler detection
+    # This initial detection does not use LDFLAGS.
     ENV.append "CXXFLAGS", "-L#{legacy_prefix}/lib"
     ENV.append "CXXFLAGS", "-L#{default_prefix}/lib"
     ENV.append "CXXFLAGS", "-lrt -lpthread" if OS.linux?
+    ENV["LD_LIBRARY_PATH"] = default_prefix/"lib"
```

Comment:
On Linux we may need to investigate setting the RPATH for `cc1`. The `LDFLAGS` are cleared when building stage1+ for `gcc`. This may have been a shotgun approach to debugging build errors.